File: LR2/27/27.ts
// Функция для проверки, является ли строка целым числом
function isInteger(str: string): boolean {
  if (typeof str !== 'string') {
    return false;
  }
  if (str.trim() === '') {
    return false;
  }
  for (let i = 0; i < str.length; i++) {
    if (str[i] < '0' || str[i] > '9') {
      return false;
    }
  }
  return true;
}

// Функция для проверки корректности даты
function isValidDate(day: number, month: number): boolean {
  return day >= 1 && day <= 31 && month >= 1 && month <= 12;
}

// Функция для обработки входной строки
function processInput(line: string): string {
  // Разбиваем введенную строку на отдельные числа
  const tokens = line.split(/\s+/);
  
  // Первое число - количество последующих чисел
  const count = parseInt(tokens[0], 10);
  if (isNaN(count)) {
    return 'Invalid input';
  }

  let result = '';

  // Обрабатываем каждое число, начиная со второго
  for (let i = 1; i <= count && i < tokens.length; i++) {
    const token = tokens[i];
    const number = parseInt(token, 10);

    if (isNaN(number)) {
      result += 'NO';
      continue;
    }

    let isCorrectDate = false;

    if (number >= 11 && number < 100) {
      const day = Math.floor(number / 10);
      const month = number % 10;
      isCorrectDate = isValidDate(day, month);
    } else if (number >= 100 && number < 1000) {
      let day = Math.floor(number / 10);
      let month = number % 10;
      isCorrectDate = isValidDate(day, month);
      
      if (!isCorrectDate) {
        day = Math.floor(number / 100);
        month = number % 100;
        isCorrectDate = isValidDate(day, month);
      }
    } else if (number >= 1000 && number <= 3112) {
      const day = Math.floor(number / 100);
      const month = number % 100;
      isCorrectDate = isValidDate(day, month);
    }

    result += isCorrectDate ? 'YES' : 'NO';
  }

  return result;
}

// Функция для получения ввода от пользователя
function getUserInput(): void {
    const input = prompt("Введите количество чисел и сами числа через пробел:");
    
    if (input === null) {
        console.log("Ввод был отменен пользователем");
        return;
    }

    const result = processInput(input);
    console.log(`Результат: ${result}`);
}

// Запускаем программу
getUserInput();

File: LR2/27/27.py
def is_valid_date(day, month):
    """
    Проверяет, является ли заданная комбинация дня и месяца корректной датой.
    """
    return day >= 1 and day <= 31 and month >= 1 and month <= 12

def check_date_format(number_str):
    """
    Проверяет, можно ли вставить символ '/' в строку так, чтобы получилась корректная дата.
    """
    if not number_str.isdigit() or len(number_str) < 2:
        return "NO"
        
    number = int(number_str)
    is_correct_date = False
    
    if 11 <= number < 100:
        day = number // 10
        month = number % 10
        is_correct_date = is_valid_date(day, month)
    elif 100 <= number < 1000:
        day = number // 10
        month = number % 10
        is_correct_date = is_valid_date(day, month)
        if not is_correct_date:
            day = number // 100
            month = number % 100
            is_correct_date = is_valid_date(day, month)
    elif 1000 <= number <= 3112:
        day = number // 100
        month = number % 100
        is_correct_date = is_valid_date(day, month)
        
    return "YES" if is_correct_date else "NO"

# Основной блок кода
input_numbers = input().split()  # Считываем последовательность чисел из ввода пользователя
n = int(input_numbers[0])  # Первое число - количество чисел для проверки
results = []

for i in range(1, n + 1):
    if i < len(input_numbers):
        results.append(check_date_format(input_numbers[i]))

print("".join(results))  